### PR TITLE
Make loose data unresolvable

### LINKED ISSUE #392: Make looseArray unresolvable.

### PR DESCRIPTION
**Pull request by reczkok**
closes #392

### CODE REVIEW

File: packages/typegpu/src/data/align.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ class AbstractTgpuAlignedImpl<
   public readonly isCustomAligned = true;
 
   constructor(
-    private data: AnyTgpuData | AnyTgpuLooseData,
+    protected data: AnyTgpuData | AnyTgpuLooseData,
```
Comment (mhawryluk) [inline packages/typegpu/src/data/align.ts]:
```suggestion
    protected data: TData,
```
that way we can eliminate checking if data is loose inside TgpuAligned

File: packages/typegpu/src/data/align.ts [hunk 2]
```diff
@@ -109,17 +109,20 @@ class AbstractTgpuAlignedImpl<
...
 export class TgpuAlignedImpl<TAlign extends number, TData extends AnyTgpuData>
   extends AbstractTgpuAlignedImpl<TAlign, TData>
   implements TgpuAligned<TAlign, TData>
 {
   public readonly isLoose = false as const;
+
+  resolve(ctx: ResolutionCtx): string {
+    if (isDataLoose(this.data)) {
+      throw new Error('Loose data cannot be resolved.');
+    }
```
Comment (mhawryluk) [inline packages/typegpu/src/data/align.ts]:
```suggestion
```

**Review by mhawryluk — CHANGES_REQUESTED** [review]

File: packages/typegpu/src/data/align.ts (hunk 1 above)
Comment (reczkok) [reply packages/typegpu/src/data/align.ts]: Great idea

**Review by iwoplaza — APPROVED** [review]
Awesome sauce! 🍝

**Pull request merged**